ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/ImageSingleVulnerabilities.tsx

```diff
@@ -28,65 +29,47 @@ import { VulnerabilitySeverity } from 'types/cve.proto';
+import useURLPagination from 'hooks/useURLPagination';
+import useURLSort, { UseURLSortProps } from 'hooks/useURLSort';
 import { getHasSearchApplied } from 'utils/searchUtils';
 import { cveStatusTabValues, FixableStatus } from './types';
 import WorkloadTableToolbar from './WorkloadTableToolbar';
 import BySeveritySummaryCard from './SummaryCards/BySeveritySummaryCard';
 import CvesByStatusSummaryCard from './SummaryCards/CvesByStatusSummaryCard';
 import SingleEntityVulnerabilitiesTable from './Tables/SingleEntityVulnerabilitiesTable';
-import useImageVulnerabilities, {
-    ImageVulnerabilitiesResponse,
-} from './hooks/useImageVulnerabilities';
+import useImageVulnerabilities from './hooks/useImageVulnerabilities';
 
-function severityCountsFromImageVulnerabilities(
-    imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities']
-): Record<VulnerabilitySeverity, number> {
-    const severityCounts = {
-        LOW_VULNERABILITY_SEVERITY: 0,
-        MODERATE_VULNERABILITY_SEVERITY: 0,
-        IMPORTANT_VULNERABILITY_SEVERITY: 0,
-        CRITICAL_VULNERABILITY_SEVERITY: 0,
-    };
-
-    imageVulnerabilities.forEach(({ severity }) => {
-        severityCounts[severity] += 1;
-    });
-
-    return severityCounts;
-}
-
-function statusCountsFromImageVulnerabilities(
-    imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities']
-): Record<FixableStatus, number> {
-    const statusCounts = {
-        Fixable: 0,
-        'Not fixable': 0,
-    };
-
-    imageVulnerabilities.forEach(({ isFixable }) => {
-        if (isFixable) {
-            statusCounts.Fixable += 1;
-        } else {
-            statusCounts['Not fixable'] += 1;
-        }
-    });
-
-    return statusCounts;
-}
+const defaultSortOptions: UseURLSortProps = {
+    sortFields: ['CVE', 'Severity', 'Fixable'],
+    defaultSortOption: {
+        field: 'Severity',
+        direction: 'desc',
+    },
+};
 
 export type ImageSingleVulnerabilitiesProps = {
     imageId: string;
 };
 
 function ImageSingleVulnerabilities({ imageId }: ImageSingleVulnerabilitiesProps) {
     const { searchFilter } = useURLSearch();
+    const { page, perPage, setPage, setPerPage } = useURLPagination(50);
+    // TODO Need to reset current page at the same time sorting changes
+    const { sortOption, getSortParams } = useURLSort(defaultSortOptions);
     // TODO Still need to properly integrate search filter with query
-    const { data, loading, error } = useImageVulnerabilities(imageId, {});
+    const pagination = {
+        offset: (page - 1) * perPage,
+        limit: perPage,
+        sortOption,
+    };
+    const { data, previousData, loading, error } = useImageVulnerabilities(imageId, {}, pagination);
 
     const [activeTabKey, setActiveTabKey] = useURLStringUnion('cveStatus', cveStatusTabValues);
 
     let mainContent: ReactNode | null = null;
 
+    const vulnerabilityData = data || previousData;
```

Comment:
Is it possible to replace `||` with `??` nullish coalescing operator (what a name)